via/services/proxy_pdf.py

```diff
@@ -0,0 +1,79 @@
+from itertools import chain
+
+import requests
+
+from via.requests_tools import add_request_headers, stream_bytes
+
+
+class ProxyPDFService:
+    TIMEOUT = 30
+
+    def iter_url(self, url, headers=None, timeout=TIMEOUT, **kwargs):
+        response = requests.get(
+            url=url,
+            headers=self.request_headers(headers),
+            stream=True,
+            timeout=timeout,
+            **kwargs
+        )
+        response.raise_for_status()
+
+        yield from stream_bytes(response)
+
+    @staticmethod
+    def response_headers(headers=None):
+        """Add headers to the response returned to the via users.
+
+        Takes an optional `headers` dictionary to add any extra headers or override the defaults.
+
+        Returns a dictionary containing the the headers.
+        """
+
+        response_headers = {
+            "Content-Disposition": "inline",
+            "Content-Type": "application/pdf",
+            # Add a very generous caching policy of half a day max-age, full
+            # day stale while revalidate.
+            "Cache-Control": "public, max-age=43200, stale-while-revalidate=86400",
+        }
+        if headers:
+            response_headers.update(headers)
+
+        return response_headers
+
+    @classmethod
+    def request_headers(cls, headers: dict = None) -> dict:
+        """Add headers to the request make to the upstream server hosting the PDF.
+
+        Takes an optional `headers` dictionary to add any extra headers or override the defaults.
+
+        Returns a dictionary containing the the headers.
+        """
+        request_headers = add_request_headers(
+            {
+                "Accept": "*/*",
+                "Accept-Encoding": "gzip, deflate",
+                "User-Agent": "(gzip)",
+            }
+        )
+        if headers:
+            request_headers.update(headers)
+        return request_headers
+
+    @staticmethod
+    def reify_first(iterable):
```

Comment:
Is this the right place for `reify_first`? 

I think you could argue it belongs to some generic `utils` package but I think this is as good as `via/views/google_drive.py`